### PR TITLE
Update __init__.py

### PR DESCRIPTION
**Pull request by AdvancedAI-NL**
I wanted to fix the problem for when the SERPs generated were less than the `num_results` specified and then the code got stuck in an infinite loop. With this new code, that shouldn't happen anymore and during my tests the code was able to keep on working as intended.

### CODE REVIEW

File: googlesearch/__init__.py [hunk 1]
```diff
@@ -2,8 +2,8 @@
 from time import sleep
 from bs4 import BeautifulSoup
 from requests import get
+import requests
```
Comment (Nv7-GitHub) [inline googlesearch/__init__.py:5]: What is the purpose of this?

File: googlesearch/__init__.py [hunk 2]
```diff
@@ -33,48 +33,47 @@ def __init__(self, url, title, description):
...
+                else:
+                    yield link["href"]
+
+            if fetched_results >= num_results:
+                break  # Stop if we have fetched the desired number of results
 
-        if start == 0:
-            return []
+        if new_results == 0:
+            print(f"Only {fetched_results} results found for query requiring {num_results} results. Moving on to the next query.")
```
Comment (Nv7-GitHub) [inline googlesearch/__init__.py:75]: Don't make a print statement

**Review by Nv7-GitHub — CHANGES_REQUESTED** [review]
Thank you for the PR and fixing this bug! Have some small changes but overall it looks good.

**Comment by AdvancedAI-NL** [comment]
> Thank you for the PR and fixing this bug! Have some small changes but overall it looks good.

You're welcome! I made a new commit fixing the problems you pointed out. Thank you for the useful code, take care!

**Review by Nv7-GitHub — APPROVED** [review]
Looks good! I'll test it out over my break.

**Pull request merged**